src/DRICH_geo.cpp

```diff
@@ -132,6 +139,17 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
     gasvolVis = vesselVis = desc.invisible();
   };
 
+  // sensor size rescaling for large sensor (catching all photons)
+  double sensorRescale = 0;
+  if (debugOpticsMode == 4) sensorRescale = 250;
+  if (sensorRescale > 0) {
+    auto seg = (CartesianGridXY) desc.readout(readoutName).segmentation();
+    seg.setGridSizeX(sensorRescale * seg.gridSizeX());
+    seg.setGridSizeY(sensorRescale * seg.gridSizeY());
+    seg.setOffsetX(sensorRescale * seg.offsetX());
+    seg.setOffsetY(sensorRescale * seg.offsetY());
```

Comment:
`seg` will go out of scope and disappear. What is its intention?

Reply:
It's created to alter the sensor grid size when creating an unrealistically large photosensor, as dd4hep crashes when running with the regular small grid size on a very large sensor. `seg` is then only needed in this scope to change those readout parameters. This fix was added by Chris, so he might have some better insight.

Reply:
Do you even get non-const access to the segmentation? I really wouldn't change the segmentation from what's in the xml.

Reply:
It's a workaround, since the sensor size is altered in this `cpp` file (from what is set in the `xml` file). We don't care about segmentation for this giant sensor, we just want to catch all the photons' real hit positions here. If we leave the segmentation as is, it will try to make 3x3 mm pixels on a giant sensor (scale of meters), overflowing the readout bit fields. Can we just turn off segmentation somehow if `debugOpticsMode == certain number(s)`? We don't run reconstruction based on this optics 'debugging' mode.

Reply:
Ok, I get this now. Resolving.